server.go

```diff
@@ -909,6 +882,11 @@ func (s *Server) newHTTP2Transport(c net.Conn, authInfo credentials.AuthInfo) tr
 		s.mu.Lock()
 		s.errorf("NewServerTransport(%q) failed: %v", c.RemoteAddr(), err)
 		s.mu.Unlock()
+		// ErrConnDispatched means that the connection was dispatched away from
+		// gRPC; those connections should be left open.
+		if err != credentials.ErrConnDispatched {
+			c.Close()
+		}
```

Comment:
This is a double-close again, isn't it?

Reply:
Here's the thing. So we have two conns this PR works on, the rawConn and the authenticated conn. The authenticated conn gets closed in NewServerTransport, and we have to deal with the RawConn closing (in the case where it's not errconndipsatched. Thus, I deleted the rawConn.Close() in regards to the handshaking, but the the authenticated conn and the raw conn are not the same logical thing, so deleting the rawConn.Close() in the error handling from transport I think solves this double close since rawConn != authenticated conn.

Reply:
> Thus, I deleted the rawConn.Close() in regards to the handshaking, but the the authenticated conn and the raw conn are not the same logical thing, so deleting the rawConn.Close() in the error handling from transport I think solves this double close since rawConn != authenticated conn.

They are not the same, but once we _have_ an authenticated conn, we no longer care about the rawConn.  The authenticated conn wraps it, and it is its job to call rawConn.Close when authenticatedConn.Close is called.